autonomy_software/planning_pkg/src/target_generator.py

```diff
@@ -212,11 +238,24 @@ def getNextWaypoint(self):
             ):
                 return wp.id
 
-        # If we are here, it means that all the waypoints behind the closest one are inside the lookahead distance
-        # in this case, we return the last waypoint
+        if self.loop_over_waypoints is True:
+            rospy.logwarn("Looping over from beginning.")
+            # If we are here, it means that all the waypoints behind the closest one are within the lookahead distance
+            # We have to loop over the waypoints starting from the beginning
+            for wp in self.waypoints.waypoints[:id_closest_wp]:
+                if (
+                    plane_distance(
+                        State(x=wp.pose.position.x, y=wp.pose.position.y),
+                        self.current_state,
+                    )
+                    > self.lookahead_distance
+                ):
+                    return wp.id
+
+        # If we are here, all the waypoints are within the lookahead distance
+        # In this case, we return the last waypoint
```

Comment:
This comment can be updated now that we don't return the last waypoint

Reply:
Done